backend/tracim_backend/lib/core/mention.py

```diff
@@ -0,0 +1,136 @@
+import typing
+
+from bs4 import BeautifulSoup
+from bs4 import Tag
+from pluggy import PluginManager
+
+from tracim_backend.config import CFG
+from tracim_backend.lib.core.content import ContentApi
+from tracim_backend.lib.core.event import BaseLiveMessageBuilder
+from tracim_backend.lib.core.event import EventApi
+from tracim_backend.lib.core.plugins import hookimpl
+from tracim_backend.lib.core.workspace import WorkspaceApi
+from tracim_backend.lib.utils.request import TracimContext
+from tracim_backend.models.data import Content
+from tracim_backend.models.event import EntityType
+from tracim_backend.models.event import OperationType
+from tracim_backend.models.event import Event
+from tracim_backend.models.tracim_session import TracimSession
+
+
+class Mention:
+    """A mention with its attributes: id and recipient."""
+
+    def __init__(self, recipient: str, id_: str) -> None:
+        self.recipient = recipient
+        self.id = id_
+
+    def __eq__(self, other) -> bool:
+        return (
+            isinstance(other, Mention) and other.recipient == self.recipient and other.id == self.id
+        )
+
+    def __repr__(self) -> str:
+        return "<Mention(recipient={}, id={})>".format(self.recipient, self.id)
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+
+class MentionBuilder:
```

Comment:
We will have in a near future mentions extracted from other content than HTML. If the builder is dedicated to HTML, I suggest to name it like this.

By the way, I suggest to prepare the design for supporting several builders (at least: markdown and html): 

- you know exactly how the backend is design for extracting mentions
- you know that other builders will be required (I think you took into account this evolution)
- you are the best player for designing a "markdown-ready" design.

Otherwise, it will require refactoring in a near future, which does not seem to be a good idea.

Reply:
In my mind `MentionBuilder` is aimed to be generic, with specific formats handled in separate classes. I hesitated to implement this directly as we currently only support html, but I can do it now.

Reply:
Changes done.